test/performance/throughput/ThroughputPublisher.cpp

```diff
@@ -760,7 +760,7 @@ bool ThroughputPublisher::load_recoveries()
     std::string line;
     size_t start;
     size_t end;
-    int32_t recovery;
+    uint32_t recovery;
```

Comment:
Doesn't seem the correct way to do this, as we are checking for negative values below (which now give additional warnings).

I would suggest adding an `int32_t input_recovery;` after the declaration of `iss` below, use it to read and validate the input, and then do a `recovery = static_cast<uint32_t>(input_recovery);` before finding and pushing to the vector.